sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,237 @@
+                             "of shape (n_features,).")
+
+        bad_nbins_value = n_bins < 2
+        is_ignored = np.zeros(n_features).astype(bool)
+        is_ignored[ignored] = True
+
+        violating_indices = np.where(~is_ignored & bad_nbins_value)[0]
+        if violating_indices.shape[0] > 0:
+            raise ValueError("Discretizer received an invalid number "
+                             "of bins at indices {}. Number of bins "
+                             "must be at least 2."
+                             .format(violating_indices.tolist()))
+        return n_bins
+
+    @staticmethod
+    def _check_ignored_features(ignored, n_features):
+        if ignored is None:
+            return np.array([], dtype='int64')
+
+        ignored = check_array(ignored, ensure_2d=False, dtype=int)
+        ignored = column_or_1d(ignored)
+
+        if len(set(ignored)) != ignored.shape[0]:
+            raise ValueError("Duplicate ignored column indices found.")
+
+        if np.all(ignored >= 0) and np.all(ignored < n_features):
+            return ignored
+
+        raise ValueError("Invalid ignored feature index.")
+
+    def transform(self, X):
+        """Discretizes the data.
+
+        Parameters
+        ----------
+        X : numeric array-like, shape (n_samples, n_features)
+            Data to be discretized.
+
+        Returns
+        -------
+        Xt : numeric array-like, shape (n_samples, n_features)
+            Data in the binned space.
+        """
+        X = self._check_X_post_fit(X)
+
+        return _transform_selected(X, self._transform,
+                                   self._transformed_features, copy=True,
+                                   retain_order=True)
+
+    def _transform(self, X):
+        """Performs transformation on X, with no ignored features."""
+        trans = self._transformed_features
+
+        X -= self.offset_[trans]
+        with np.errstate(divide='ignore', invalid='ignore'):
+            X //= self.bin_width_[trans]
+
+        X[~np.isfinite(X)] = 0
+        np.clip(X, 0, self.n_bins_[trans] - 1, out=X)
+        return X
+
+    def inverse_transform(self, Xt):
+        """Given transformed (binned) data, returns a representation of
+        the data in the original feature space.
+
+        Parameters
+        ----------
+        Xt : numeric array-like, shape (n_sample, n_features)
+            Transformed data in the binned space.
+
+        Returns
+        -------
+        Xinv : numeric array-like
+            Data in the original feature space.
+
+        """
+        Xt = self._check_X_post_fit(Xt)
+        trans = self._transformed_features
+        Xinv = Xt.copy()
+        Xinv_sel = Xinv[:, trans]
+
+        Xinv_sel *= self.bin_width_[trans]
+        Xinv_sel += self.offset_[trans]
+        Xinv_sel += self.bin_width_[trans] / 2
+
+        Xinv[:, trans] = Xinv_sel
+        return Xinv
+
+    def _check_X_post_fit(self, X):
```

Comment:
Do you mean the definition of the function itself? 

I personally like the name of the function, since it communicates that 

1. We are checking `X`, and no other variables
2. This check is only invoked after fitting the transformer

Of course, I'm open to ideas. If there's a more elegant name which communicates the two above ideas, I'm willing to take it.